internal/command/state_identities.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/cli"
+	"github.com/hashicorp/terraform/internal/addrs"
+	"github.com/hashicorp/terraform/internal/states"
+	"github.com/hashicorp/terraform/internal/tfdiags"
+)
+
+// StateIdentitiesCommand is a Command implementation that lists the resource identities
+// within a state file.
+type StateIdentitiesCommand struct {
```

Comment:
I think we also need to register this command, to make it runnable from CLI.

Somewhere around here: https://github.com/hashicorp/terraform/blob/7cbc9d00cf7eb5a873de968c54af7e2c67d5e5a6/commands.go#L371-L379